docs/src/main/sphinx/release/release-431.md

```diff
@@ -0,0 +1,60 @@
+# Release 431 (27 Oct 2023)
+
+## General
+
+* Add support for [](/routines). ({issue}`19308`)
+* Add support for [](/sql/create-function) and [](/sql/drop-function) statements. ({issue}`19308`)
+* Add support for the `REPLACE` modifier to the `CREATE TABLE` statement. ({issue}`13180`)
+* Disallow a `null` offset for the {func}`lead` and {func}`lag` functions. ({issue}`19003`)
+* Improve performance of queries with short running splits. ({issue}`19487`)
+
+## Security
+
+* Support defining rules for procedures in file-based access control. ({issue}`19416`)
+* Mask additional sensitive values in log files. ({issue}`19519`)
+
+## JDBC driver
+
+* Allow configuration of executing statements with the legacy
+  `PREPARE...EXECUTE` or the faster `EXECUTE IMMEDIATE` invocation.
+  ({issue}`19541`)
+
+## Delta Lake connector
+
+* Replace the `hive.metastore-timeout` Hive metastore configuration property
+  with the `hive.metastore.thrift.client.connect-timeout` and
+  `hive.metastore.thrift.client.read-timeout` properties. ({issue}`19390`)
+
+## Hive connector
+
+* Add support for [SQL routine management](sql-routine-management). ({issue}`19308`)
+* Replace the `hive.metastore-timeout` Hive metastore configuration property
+  with the `hive.metastore.thrift.client.connect-timeout` and
+  `hive.metastore.thrift.client.read-timeout` properties. ({issue}`19390`)
+* Cast char fields, if necessary, to varchar type in Hive view translations. ({issue}`18337`)
```

Comment:
That's the implementation detail. What problem is this solving?

Reply:
Prevents query translations failures.. updated